src/software/ai/navigator/obstacle/const_velocity_obstacle_test.cpp

```diff
@@ -0,0 +1,73 @@
+#include "software/ai/navigator/obstacle/const_velocity_obstacle.hpp"
+
+#include <gtest/gtest.h>
+
+static constexpr double MAX_OBSTACLE_TIME_HORIZON = 1.0;
+
+TEST(ConstVelocityObstacleTest, circle_obstacle_contains)
+{
+    Circle circle({2, 0}, 1);
+    ObstaclePtr obstacle(std::make_shared<ConstVelocityObstacle<Circle>>(
+        circle, Vector(0.5, 0), MAX_OBSTACLE_TIME_HORIZON));
+    // Point barely inside obstacle
+    Point point_1(1.01, 0);
+    // Point barely outside obstacle
+    Point point_2(3.01, 0);
+
+    // 0 seconds into the future
+    EXPECT_TRUE(obstacle->contains(point_1, 0.0));
+    EXPECT_FALSE(obstacle->contains(point_2, 0.0));
+
+    // After 1 second, the obstacle should have moved to (2.5, 2)
+    EXPECT_FALSE(obstacle->contains(point_1, 1.0));
+    EXPECT_TRUE(obstacle->contains(point_2, 1.0));
+
+    // After >MAX_OBSTACLE_TIME_HORIZON seconds, we should not predict the obstacles
+    // position
+    EXPECT_FALSE(obstacle->contains(point_1, MAX_OBSTACLE_TIME_HORIZON + 1.0));
+    EXPECT_TRUE(obstacle->contains(point_2, MAX_OBSTACLE_TIME_HORIZON + 1.0));
```

Comment:
if we don't predict the obstacle's positions, dont we expect both of these to return FALSE?

Reply:
Oh I understand now, it only predicts up to MAX_OBSTACLE_TIME_HORIZON